test/org/sagebionetworks/bridge/validators/StudyValidatorTest.java

```diff
@@ -210,5 +214,31 @@ public void cannotSetMaxNumOfParticipantsLessThanZero() {
         study.setMaxNumOfParticipants(-100);
         assertCorrectMessage(study, "maxNumOfParticipants", "maxNumOfParticipants must be zero (no limit on enrollees) or higher");
     }
+    
+    @Test
+    public void shortListOfDataGroupsOK() {
+        study.setDataGroups(Sets.newHashSet("beta_users", "production_users", "testers", "internal"));
+        Validate.entityThrowingException(StudyValidator.INSTANCE, study);
+    }
+    
+    @Test
+    public void longListOfDataGroupsInvalid() {
+        study.setDataGroups(newLinkedHashSet("Antwerp", "Ghent", "Charleroi", "Liege", "Brussels-City", "Bruges", "Schaerbeek", "Anderlecht", "Namur", "Leuven", "Mons", "Molenbeek-Saint-Jean"));
+        assertCorrectMessage(study, "dataGroups", "dataGroups will not export to Synapse (string is over 100 characters: 'Antwerp, Ghent, Charleroi, Liege, Brussels-City, Bruges, Schaerbeek, Anderlecht, Namur, Leuven, Mons, Molenbeek-Saint-Jean')");
+    }
+    
+    @Test
+    public void dataGroupCharactersRestricted() {
+        study.setDataGroups(Sets.newHashSet("Liège"));
+        assertCorrectMessage(study, "dataGroups", "dataGroups contains invalid tag 'Liège' (only letters, numbers, underscore and dash allowed)");
+    }
+    
+    private Set<String> newLinkedHashSet(String... list) {
+        Set<String> set = new LinkedHashSet<>();
+        for (String element : list) {
+            set.add(element);
+        }
+        return set;
+    }
```

Comment:
Minor: Might I suggest a TreeSet instead? TreeSets are naturally ordered, and the TreeSet constructor takes a collection, so you won't need this helper method.